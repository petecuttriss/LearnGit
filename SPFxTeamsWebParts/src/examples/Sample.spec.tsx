import * as React from 'react';

import 'jest';
import { createRenderer } from 'react-test-renderer/shallow';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DisplayMode } from '@microsoft/sp-core-library';
import { Sample, ISampleProps} from './Sample';

//jest.mock('@microsoft/sp-core-library/', () => 'BrowserDetection');
jest.mock('@microsoft/sp-core-library/', () => {
  return {
    'BrowserDetection': {},
    'DisplayMode': {}
  };
});
jest.mock('@pnp/spfx-controls-react/lib/WebPartTitle', () => {
  return {
    'WebPartTitle': 'WebPartTitle'
  };
});

test('should render Sample component correctly', () => {
  /*
   * using the OOTB 'react-test-renderer'
   */
  /*const renderer = createRenderer();
  renderer.render(<Sample />);
  console.log(renderer.getRenderOutput());
  expect(renderer.getRenderOutput()).toMatchSnapshot();*/

  /*
   * using enzyme
   */
  
  const wrapper = shallow(React.createElement(Sample, {}));
  expect(wrapper.find('span.message').text()).toBe('Hello world:');
  expect(wrapper.find('li').length).toBe(3);

  expect(toJson(wrapper)).toMatchSnapshot();
  expect(wrapper).toMatchSnapshot();
});