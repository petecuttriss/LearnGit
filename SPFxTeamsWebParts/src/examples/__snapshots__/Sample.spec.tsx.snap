// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Sample component correctly 1`] = `
<div>
  <span
    className="message"
  >
    Hello world:
  </span>
  <ul>
    <li>
      one
    </li>
    <li>
      two
    </li>
    <li>
      three
    </li>
  </ul>
  <StyledMessageBarBase
    messageBarType={0}
  >
    No results found.
  </StyledMessageBarBase>
  <StyledSearchBoxBase />
  <Placeholder
    buttonLabel="Configure"
    description="Please configure the web part."
    iconName="Edit"
    iconText="Configure your web part"
  />
  <WebPartTitle
    title="Hello"
    updateProperty={[Function]}
  />
  <StyledNavBase
    collapsedStateText="Collapsed"
    expandedStateText="Expand"
    groups={
      Array [
        Object {
          "links": Array [],
        },
      ]
    }
    key="umqj044qya1krec1b6snn"
  />
</div>
`;

exports[`should render Sample component correctly 2`] = `
<div>
  <span
    className="message"
  >
    Hello world:
  </span>
  <ul>
    <li>
      one
    </li>
    <li>
      two
    </li>
    <li>
      three
    </li>
  </ul>
  <StyledMessageBarBase
    messageBarType={0}
  >
    No results found.
  </StyledMessageBarBase>
  <StyledSearchBoxBase />
  <Placeholder
    buttonLabel="Configure"
    description="Please configure the web part."
    iconName="Edit"
    iconText="Configure your web part"
  />
  <WebPartTitle
    title="Hello"
    updateProperty={[Function]}
  />
  <StyledNavBase
    collapsedStateText="Collapsed"
    expandedStateText="Expand"
    groups={
      Array [
        Object {
          "links": Array [],
        },
      ]
    }
    key="umqj044qya1krec1b6snn"
  />
</div>
`;
